olp-cpp-sdk-dataservice-read/src/Common.h

```diff
@@ -86,7 +86,27 @@ inline client::CancellationToken AddTask(
   repository::ExecuteOrSchedule(task_scheduler, [=] {
     context.Execute();
     pending_requests->Remove(context);
-  });
+  }, thread::NORMAL);
+
+  return context.CancelToken();
+}
+
+template <typename Function, typename Callback, typename... Args>
+inline client::CancellationToken AddTaskWithPriority(
```

Comment:
Is it not possible to also name this AddTask() and forward the first one to this, e.g.

```cpp
template <typename Function, typename Callback, typename... Args>
inline client::CancellationToken AddTask(
    const std::shared_ptr<thread::TaskScheduler>& task_scheduler,
    const std::shared_ptr<client::PendingRequests>& pending_requests,
    Function task, Callback callback, Args&&... args) {
      return AddTask(task_scheduler, pending_requests, std::move(task), std::move(callback), thread::NORMAL, std::forward<Args>(args)...);
}
```